qualtran/bloqs/util_bloqs.py

```diff
@@ -50,8 +50,31 @@
     from qualtran.simulation.classical_sim import ClassicalValT
 
 
+class UtilBloq(Bloq, metaclass=abc.ABCMeta):
+    """Base class for util bloqs.
+
+    1. Trivial controlled - pass through the control register.
+    2. Do not affect T complexity.
+    """
+
+    def get_ctrl_system(
```

Comment:
Would this me more appropriate as a mixin similar to SpecializedSingleQubitControlledGate?

Reply:
Right now they are equivalent. The only difference is that `SpecializedSingleQubitControlledGate` has a method marked as `abc.abstractmethod` (i.e. `control_registers`), so it seems we don't need to explicitly specify `metaclass=abc.ABCMeta`. But here we have to.

Reply:
to be clear: they're not equivalent. This implementation _ignores_ ctrl soquets whereas the other will plug in a single-bit control to an expected location

Reply:
Ah I just meant they are equivalent in design, i.e. not a mixin but a subclass of Bloq